pennylane/gradients/pulse_gradient.py

```diff
@@ -0,0 +1,529 @@
+        sample_seed (int): randomness seed to be used for the time samples in the stochastic
+            parameter-shift rule
+        shots (None, int, list[int]): Argument used by the new return type system
+            (see :func:`~.enable_return` for more information); it represents the device shots
+            that will be used to execute the tapes outputted by this transform. Note that this
+            argument doesn't influence the shots used for tape execution, but provides information
+            to the transform about the device shots and helps in determining if a shot sequence
+            was used.
+
+    Returns:
+        function or tuple[list[QuantumTape], function]:
+
+        - If the input is a QNode, an object representing the Jacobian (function) of the QNode
+          that can be executed to obtain the Jacobian matrix.
+          The type of the matrix returned is either a tensor, a tuple or a
+          nested tuple depending on the nesting structure of the original QNode output.
+
+        - If the input is a tape, a tuple containing a
+          list of generated tapes, together with a post-processing
+          function to be applied to the results of the evaluated tapes
+          in order to obtain the Jacobian matrix.
+
+    This transform realizes the stochastic parameter-shift rule for pulse sequences, as introduced
+    in `Banchi and Crooks (2018) <https://quantum-journal.org/papers/q-2021-01-25-386/>`_ and
+    `Leng et al. (2022) <https://arxiv.org/abs/2210.15812>`_.
+
+    .. note::
+
+        This function requires the JAX interface and does not work with other autodiff interfaces
+        commonly encountered with PennyLane. It additionally only supports the new return type
+        system, see :func:`enable_return`.
+
+    .. note::
+
+        This function uses a basic sampling approach with a uniform distribution to estimate the
+        integral appearing in the stochastic parameter-shift rule. In many cases, there are
+        probability distributions that lead to smaller variances of the estimator.
+        In addition, the sampling approach will not reduce trivially to simpler parameter-shift
+        rules when used with simple pulses (see details and examples below), potentially leading
+        to imprecise results and/or unnecessarily large computational efforts.
+
+    .. note::
+
+        Currently this function only supports pulses for which each *parametrized* term is a
+        simple Pauli word. More general Hamiltonian terms are not supported yet.
+
+    **Examples**
+
+    Consider a pulse program with a single two-qubit pulse, generated by a Hamiltonian
+    with three terms: the non-trainable term :math:`\frac{1}{2}X\otimes I`, the trainable
+    constant (over time) term :math:`v_1 Z\otimes Z` and the trainable sinoidal term
+    :math:`\sin(v_2 t) I\otimes X`.
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit.jax", wires=2)
```

Comment:
With interfaces it will unwrap before going to "default.qubit.jax" is it something that we would like to avoid. Maybe it is changed, but I remember that "default.qubit.jax" explicitly says that it only supports backprop, if it is the case it is worth to change the doc.

Reply:
Oh, good catch! 
As far as I am aware, pulse programming requires `default.qubit.jax` to be set.
So yeah, probably necessary to modify the device doc.

Reply:
I added a statement to the `default.qubit.jax` docstring regarding its `analytic` argument (not sure this is the best place?).
There is the line 
```python
f"Device {device.short_name} only supports diff_method='backprop' when using the "
```
in the QNode code [(l.649)](https://github.com/PennyLaneAI/pennylane/blob/deb6ae380a3baedb5045dbd27f098c346b09d2c5/pennylane/qnode.py#L649)
but I'm not sure whether we want to add any specific information there, as it only applies to the JAX device and this error is actually not raised when using the device with `shots!=None`.

Reply:
Yes I think it is fine to keep it only in the jax specific device.